controllers/nnf_workflow_controller.go

```diff
@@ -840,18 +859,18 @@ func (r *NnfWorkflowReconciler) startPreRunState(ctx context.Context, workflow *
 		}
 	}
 
+	readyResult, err := r.waitForNnfAccessStateAndReady(ctx, workflow, index, "mounted")
```

Comment:
Why did this get moved from finishPreRunState?

Reply:
I originally moved to this to startPreRunState so that when in finishPreRunState, we'd ensure that the NNFAccesses are mounted before we start checking to see if the containers started. But looking at this again, I don't think that is necessary. I'll put it back and test.

Reply:
I remember now: keeping this in finishPreRunState at the top of the block ends up in an error because container directives do not have any NnfAccess. So I moved it to the end of startPreRunState because only non-container directives get that far - container directives return before doing anything with NnfAccess.

So I could keep this here or perhaps move it further down after `return r.waitForContainersToStart(ctx, workflow, index)`.

Reply:
@matthew-richerson and I discussed this, and I kept this in `finishPreRunState()` and moved it down after the switch block where `r.waitForContainersToStart()` would return for container directives.